lib/fog/aws/data_pipeline.rb

```diff
@@ -116,7 +167,19 @@ def request(params)
         end
 
         def _request(params)
-          @connection.request(params)
+          response = @connection.request(params)
+
+          unless response.body.empty?
```

Comment:
@lanej Won't this part mean that we still end up with decoded bodies? Or am I misunderstanding your question?